graylog2-web-interface/src/components/common/SearchForm.jsx

```diff
@@ -88,27 +103,37 @@ class SearchForm extends React.Component {
   _onSearch = (e) => {
     e.preventDefault();
 
-    this._setLoadingState();
-    this.props.onSearch(this.refs.query.value, this._resetLoadingState);
+    this._setLoadingState().then(() => {
+      this.props.onSearch(this.state.query, this._resetLoadingState);
+    });
   };
 
   _onReset = () => {
     this._resetLoadingState();
-    this.refs.query.value = '';
-    this.props.onReset();
+    this.setState({ query: '' }, this.props.onReset);
+  };
+
+  handleQueryChange = (e) => {
+    const query = e.target.value;
+    this.setState({ query: query }, () => {
```

Comment:
Is there any reason to wait for the state to be updated before calling `this.props.onQueryChange()`? The prop is in the end not getting anything from the state, so if this is really needed we should add a comment.